client/blocks/importer/wordpress/import-everything/import-users/index.tsx

```diff
@@ -226,47 +359,41 @@ const ImportUsers = ( { site, onSubmit }: Props ) => {
 	};
 
 	useEffect( () => {
-		let filteredUserData = usersData;
-		// We need to remove the current admin user from the list
-		if ( userId && filteredUserData?.users ) {
-			filteredUserData = {
-				...filteredUserData,
-				users: filteredUserData?.users?.filter( ( userItem ) => {
-					return userItem?.linked_user_ID !== userId;
-				} ),
-			};
-		}
-
-		const updatedUsers = filteredUserData?.users?.map( ( userItem ) => userItem );
-		const storedUsers = usersList?.map( ( userItem ) => userItem.user );
-
-		if ( JSON.stringify( updatedUsers ) !== JSON.stringify( storedUsers ) ) {
-			let updatedUsersList = updatedUsers?.map( ( user ) => ( { user, checked: true } ) ) || [];
-			const storedUsersList = usersList;
-
-			if ( userId && updatedUsersList ) {
-				updatedUsersList = updatedUsersList.filter( ( userItem ) => {
-					return userItem?.user?.linked_user_ID !== userId;
-				} );
-			}
-
-			// Update the checked status of the users to maintain the correct value
-			updatedUsersList = updatedUsersList?.map( ( userItem ) => {
-				const userId = userItem?.user?.ID;
-				const userExists = storedUsersList?.find(
-					( storedUser ) => storedUser?.user?.ID === userId
+		let updatedUsersList = [ ...usersList ];
+		if ( userListFilters.searchQuery ) {
+			updatedUsersList = updatedUsersList.filter( ( user ) => {
+				const userEmail =
+					( typeof user.user?.email === 'string' && user.user?.email?.toLowerCase() ) || '';
+				const userLogin = user.user?.login?.toLowerCase();
+				const userName = user.user?.name?.toLowerCase();
+				const userNiceName = user.user?.nice_name?.toLowerCase();
+				const userQuery = userListFilters.searchQuery.toLowerCase();
+				const currentAdmin = user.user?.linked_user_ID === userId;
+
+				return (
+					! currentAdmin &&
+					( userEmail.includes( userQuery ) ||
+						userLogin.includes( userQuery ) ||
+						userName.includes( userQuery ) ||
+						userNiceName.includes( userQuery ) )
 				);
-				userItem.checked = userExists ? userExists?.checked : true;
-
-				return userItem;
 			} );
+		}
 
-			const updatedCheckedUsersNumber = updatedUsersList?.filter( ( x ) => x.checked )?.length;
+		if ( userListFilters.selectedRoleFilters.length ) {
+			updatedUsersList = updatedUsersList.filter( ( user ) => {
+				const userRoles = user.user?.roles;
```

Comment:
Are we able to use `getRole` here?

Reply:
Yes, `getRole` will work here. Updating this.